src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
I'm of the mind that overloading the meaning of arguments makes understanding usage harder, and leads to difficult to solve bugs. I'm actually already planning in Diactoros to remove the ability to pass either a string filename or resource to `Stream`, in favor of always passing a resource.

As such, I'm on board with these suggestions from @mindplay-dk:

- `ResourceStreamFactory::createResourceStream(resource $handle);`
- `TempStreamFactory::createTempStream(string $content = "");`

I don't see any reason for the `FileStreamFactory`; it can already be accomplished with:

```php
$factory = new ResourceStreamFactory();
$stream = $factory->createResourceStream(fopen($filename, $mode));
```

which shares the same level of explicitness at the code level, and none of the testing nightmares. (Seriously: I can use vfsStream, but I'm much happier when I don't need to. Passing `fopen('php://temp', 'w+')` is far easier!)